src/libutil/source-path.hh

```diff
@@ -11,6 +11,22 @@
 
 namespace nix {
 
+enum class SymlinkResolution {
```

Comment:
```suggestion
/**
 * Note there is a decent chance this type soon goes away because the problem is solved another way.
 */
enum class SymlinkResolution {
```